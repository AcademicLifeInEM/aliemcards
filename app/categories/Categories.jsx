import React, { PureComponent, PropTypes } from 'react';
import { post } from 'axios';
import { Link } from 'react-router-dom';

export default class Categories extends PureComponent {

  constructor(props) {
    super(props);
    this.state = {
      categories: [],
      loading: true,
    };
  }

  componentDidMount() {
    post('/graphql', {
      query: `query {
        categories {
          id
          name
        }
      }`,
    })
    .then(res => {
      if (res.status !== 200) throw res.status;
      const { categories } = res.data.data;
      this.setState({ categories, loading: false });
    })
    .catch(err => console.error(`Error: API response status code = ${err}`));
  }

  render() {
    return (
      <div>
        <h1>Categories</h1>
        <Results categories={this.state.categories} />
      </div>
    );
  }
}

const Results = ({ categories }) =>
  <div className="card-list">
    {categories.map(category => (
      <div key={category.id} className="card-list__item">
        <Link to={`/categories/${category.id}`} className="card-list__item-title">{category.name}</Link>
      </div>
      )
    )}
  </div>;

Results.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  })),
};
