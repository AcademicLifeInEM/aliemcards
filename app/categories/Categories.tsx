import * as React from 'react';
import { gql, graphql } from 'react-apollo';
import { Link } from 'react-router-dom';

export interface Category {
  id: string;
  name: string;
}

interface Props {
  data: {
    categories: Category[];
  };
}

const categoryQuery = gql`
  query allCategories {
    categories {
      id
      name
    }
  }
`;

@graphql(categoryQuery)
export default class Categories extends React.PureComponent<Props, {}> {
  render() {
    const categories = this.props.data.categories || [];
    return (
      <div>
        <h1>Categories</h1>
        <div className="card-list">
          {
            categories.map(category => (
              <div key={category.id} className="card-list__item">
                <Link to={`/categories/${category.id}`} className="card-list__item-title">{category.name}</Link>
              </div>
            ))
          }
        </div>
      </div>
    );
  }
}
