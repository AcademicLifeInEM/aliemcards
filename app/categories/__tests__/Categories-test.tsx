jest.mock('react-apollo');
jest.mock('react-router-dom');

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Categories from '../Categories';

const mockData = [
  {
    id: 'test-category-1',
    name: 'Test Category 1',
  },
  {
    id: 'test-category-2',
    name: 'Test Category 2',
  },
];

const setup = (categories: any = [...mockData]) => {
  const data = {
    categories,
  };
  const snapshot = renderer.create(<Categories data={data} />);
  return {
    snapshot,
  };
};

describe('<Categories />', () => {
  it('should render appropriately', () => {
    const { snapshot: withCategories } = setup();
    expect(withCategories.toJSON()).toMatchSnapshot();

    const { snapshot: withoutCategories } = setup(null);
    expect(withoutCategories.toJSON()).toMatchSnapshot();
  });
});
