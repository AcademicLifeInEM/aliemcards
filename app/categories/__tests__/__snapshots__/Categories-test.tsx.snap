// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> should render appropriately 1`] = `
<div>
  <h1>
    Categories
  </h1>
  <div
    className="card-list"
  >
    <div
      className="card-list__item"
    />
    <div
      className="card-list__item"
    />
  </div>
</div>
`;

exports[`<Categories /> should render appropriately 2`] = `
<div>
  <h1>
    Categories
  </h1>
  <div
    className="card-list"
  />
</div>
`;
